### PR TITLE
chore(build): upgrade Go to 1.19

### LINKED ISSUE #2768: Go 1.19 support
Hello 😊 

## Issue summary

- compiling gossamer should work with the latest go version
- however, the dependency https://github.com/lucas-clemente/quic-go does not support 1.19 yet

```
~/.opt/ChainSafe/gossamer development 12s
❯ go version
go version go1.19 linux/amd64
```

## Other information and links

```
# github.com/lucas-clemente/quic-go/internal/qtls
../../../.go/pkg/mod/github.com/lucas-clemente/quic-go@v0.27.1/internal/qtls/go119.go:6:13: cannot use "The version of quic-go you're using can't be built on Go 1.19 yet. For more details, please see https://github.com/lucas-clemente/quic-go/wiki/quic-go-and-Go-versions." (untyped string constant "The version of quic-go you're using can't be built on Go 1.19 yet. F...) as int value in variable declaration
make: *** [Makefile:73: build] Error 2
```

How are you managing go versions usually?

### PR DESCRIPTION
**Pull request by qdm12**
## Changes

- CI uses Go 1.19
- Dockerfiles use Go 1.19
- go.mod uses Go 1.19
- Readme: remove Go 1.19 incompatibility warning
- Readme: update requirement to Go 1.19 or higher
- Re-generate proto files to match Go 1.19 new comments format
- Fix all comments to match Go 1.19 new comments format
- Replace `ioutil` import with `os`
- Fix new lint errors introduced
- Remove old `//gocyclo:ignore` comment

## Issues

Fixes #2768 

## Primary Reviewer

@EclesioMeloJunior

### CODE REVIEW

**Review by EclesioMeloJunior — APPROVED** [review]

File: dot/rpc/modules/sync_state_integration_test.go [hunk 1]
```diff
@@ -19,7 +19,7 @@ import (
 func TestSyncStateModule(t *testing.T) {
 	fp := utils.GetWestendDevRawGenesisPath(t)
 
-	data, err := ioutil.ReadFile(filepath.Clean(fp))
+	data, err := os.ReadFile(filepath.Clean(fp))
 	require.NoError(t, err)
```
Comment (kishansagathiya) [inline dot/rpc/modules/sync_state_integration_test.go:24]: why?

File: lib/runtime/allocator.go [hunk 2]
```diff
@@ -40,14 +40,14 @@ type FreeingBumpHeapAllocator struct {
...
 //
-// * `ptrOffset` - The pointers returned by `Allocate()` start from this
-//   offset on. The pointer offset needs to be aligned to a multiple of 8,
-//   hence a padding might be added to align `ptrOffset` properly.
+//   - `ptrOffset` - The pointers returned by `Allocate()` start from this
+//     offset on. The pointer offset needs to be aligned to a multiple of 8,
+//     hence a padding might be added to align `ptrOffset` properly.
 //
-// * returns a pointer to an initilized FreeingBumpHeapAllocator
+//   - returns a pointer to an initilized FreeingBumpHeapAllocator
```
Comment (kishansagathiya) [inline lib/runtime/allocator.go:50]: ?

File: lib/transaction/types.go [hunk 3]
```diff
@@ -50,24 +50,30 @@ func NewValidTransaction(extrinsic types.Extrinsic, validity *Validity) *ValidTr
...
+  - `Usurped`
+  - `Dropped`
+
+4. Re-entering the pool:
+  - `Retracted`
+
+5. Block finalized:
+  - `Finalized`
+  - `FinalityTimeout`
+*/
```
Comment (kishansagathiya) [inline lib/transaction/types.go:76]: ?

File: tests/rpc/rpc_01-system_test.go [hunk 4]
```diff
@@ -111,8 +111,8 @@ func TestSystemRPC(t *testing.T) { //nolint:tparallel
 
 		expectedResponse := modules.SystemPeersResponse{
 			// Assert they all have the same best block number and hash
-			{Roles: 4, PeerID: ""},
-			{Roles: 4, PeerID: ""},
+			common.PeerInfo{Roles: 4, PeerID: ""},
+			common.PeerInfo{Roles: 4, PeerID: ""},
```
Comment (kishansagathiya) [inline tests/rpc/rpc_01-system_test.go:115]: why ?

**Review by kishansagathiya — COMMENTED** [review]
Upgrade to Go 1.19 is good.

Not directly relevant, but there are a lot of space removals, which is alright. Are they by-product of linter?

but I don't see the point of some the other changes that I have put ? on.

File: dot/rpc/modules/sync_state_integration_test.go (hunk 1 above)
Comment (qdm12) [reply dot/rpc/modules/sync_state_integration_test.go:24]: Because ioutil is deprecated since go 1.17

File: lib/transaction/types.go (hunk 3 above)
Comment (qdm12) [reply lib/transaction/types.go:76]: gofmt

File: tests/rpc/rpc_01-system_test.go (hunk 4 above)
Comment (qdm12) [reply tests/rpc/rpc_01-system_test.go:115]: linter complaining

Comment (kishansagathiya) [reply tests/rpc/rpc_01-system_test.go:115]: tried running `make lint`, did not throw error. But I see the point of making it explicit that it is common.PeerInfo.

How about 
```
		expectedResponse := modules.SystemPeersResponse(
			[]common.PeerInfo{
				// Assert they all have the same best block number and hash
				{Roles: 4, PeerID: ""},
				{Roles: 4, PeerID: ""},
			},
		)
```

Comment (kishansagathiya) [reply tests/rpc/rpc_01-system_test.go:115]: just ran `make lint` after install go 1.19. did not complain.

**Review by kishansagathiya — APPROVED** [review]

File: tests/rpc/rpc_01-system_test.go (hunk 4 above)
Comment (qdm12) [reply tests/rpc/rpc_01-system_test.go]: Oh well nothing complains anymore, I guess mystery unresolved, conversation resolved!

**Pull request merged**